src/main/java/io/github/noeppi_noeppi/libx/data/provider/SoundDefinitionsProviderBase.java

```diff
@@ -0,0 +1,154 @@
+package io.github.noeppi_noeppi.libx.data.provider;
+
+import io.github.noeppi_noeppi.libx.mod.ModX;
+import net.minecraft.data.DataGenerator;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.sounds.SoundEvent;
+import net.minecraftforge.common.data.ExistingFileHelper;
+import net.minecraftforge.common.data.SoundDefinition;
+import net.minecraftforge.common.data.SoundDefinitionsProvider;
+import net.minecraftforge.registries.ForgeRegistries;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * A base class for sound definition provider. Adds methods to add multiple sounds with same name at once.
+ */
+public abstract class SoundDefinitionsProviderBase extends SoundDefinitionsProvider {
+
+    protected final ModX mod;
+
+    private final Set<SoundEvent> blacklist = new HashSet<>();
+
+    public SoundDefinitionsProviderBase(ModX mod, DataGenerator generator, ExistingFileHelper helper) {
+        super(generator, mod.modid, helper);
+        this.mod = mod;
+    }
+
+    @Nonnull
+    @Override
+    public String getName() {
+        return this.mod.modid + " sound definitions";
+    }
+
+    /**
+     * This sound event will not be processed by the default generator
+     */
+    protected void manualSound(SoundEvent soundEvent) {
```

Comment:
There is only one sound json, so you don't do that sound *manually*. Maybe this should be renamed to something like `ignore`